src/DynamoCore/Graph/Nodes/NodeInputData.cs

```diff
@@ -0,0 +1,148 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Graph.Nodes
+{
+    public enum NodeInputTypes
+    {
+        numberInput, booleanInput, stringInput, colorInput, dateInput, selectionInput
+    };
+
+
+
+    /// <summary>
+    /// Represents a node which acts as a UI input for the graph
+    /// - may also hold a value for that input
+    /// </summary>
+    public class NodeInputData
+    {
+        /// <summary>
+        /// The id of the node.
+        /// </summary>
+        public string Id { get; set; }
+        /// <summary>
+        /// Display name of the input node.
+        /// </summary>
+        public string Name { get; set; }
+        /// <summary>
+        /// The type of input this node is.
+        /// </summary>
+        [JsonConverter(typeof(NodeInputTypeConverter))]
+        public NodeInputTypes Type { get; set; }
+        /// <summary>
+        /// The value of the input when the graph was saved.
+        /// </summary>
+        public string Value { get; set; }
+
+        //optional properties, might be null
+        /// <summary>
+        /// If this input is a selection type a list of choices a user can select.
+        /// </summary>
+        [JsonProperty(NullValueHandling=NullValueHandling.Ignore)]
+        public List<string> Choices { get; set; }
```

Comment:
If I understand correctly, the out put of this property will be ignored when serialized when null. Can you update the output you put in the PR description?